src/ngx_http_lua_string.c

```diff
@@ -433,17 +433,18 @@ ngx_http_lua_ffi_unescape_uri(const u_char *src, size_t len, u_char *dst)
 
 
 size_t
-ngx_http_lua_ffi_uri_escaped_length(const u_char *src, size_t len)
+ngx_http_lua_ffi_uri_escaped_length(const u_char *src, size_t len,
+    int type)
 {
-    return len + 2 * ngx_http_lua_escape_uri(NULL, (u_char *) src, len,
-                                             NGX_ESCAPE_URI_COMPONENT);
+    return len + 2 * ngx_http_lua_escape_uri(NULL, (u_char *) src, len, type);
```

Comment:
We should validate if the type is out of range...it could be any integer value anyway.